Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,437 @@
+
+				Crecord?.Dispose();
+				Crecord = null;
+
+				Clist?.Dispose();
+				Clist = null;
+
+				Cmanager?.Dispose();
+				Cmanager = null;
+
+				return calendars;
+			});
+		}
+
+		/// <summary>
+		/// Gets a single calendar event by platform-specific ID.
+		/// </summary>
+		/// <param name="externalId">Platform-specific calendar event identifier</param>
+		/// <returns>The corresponding calendar event, or null if not found</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<CalendarEvent> GetEventByIdAsync(string externalId)
+		{
+			CalendarEvent events = null;
+			CalendarManager Cmanager = new CalendarManager();
+			CalendarList Clist = Cmanager.Database.GetAll(Event.Uri, 0, 0);
+
+			if (Clist.Count == 0)
+				return Task.Run(() => events);
+
+			Clist.MoveLast();
+
+			TimeSpan utc = new TimeSpan(9, 0, 0);
+			CalendarRecord Crecord = Clist.GetCurrentRecord();
+
+			string Summary = Crecord.Get<string>(Event.Summary);
+			int IsAllday = Crecord.Get<int>(Event.IsAllday);
+			CalendarTime Start = Convert.ToBoolean(IsAllday)
+							? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
+							: Crecord.Get<CalendarTime>(Event.Start);
+			CalendarTime End = Convert.ToBoolean(IsAllday)
+							? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
+							: Crecord.Get<CalendarTime>(Event.End);
+			string Location = Crecord.Get<string>(Event.Location);
+			string Description = Crecord.Get<string>(Event.Description);
+			int Id = Crecord.Get<int>(Event.Id);
+
+			Crecord?.Dispose();
+			Crecord = null;
+
+			Clist?.Dispose();
+			Clist = null;
+
+			Cmanager?.Dispose();
+			Cmanager = null;
+
+			events = new CalendarEvent
+			{
+				Name = Summary,
+				Start = Start.UtcTime + utc,
+				End = End.UtcTime + utc,
+				Location = Location,
+				Description = Description,
+				AllDay = Convert.ToBoolean(IsAllday),
+				ExternalID = Id.ToString(),
+			};
+
+			return Task.Run(() => events);
+		}
+
+		/// <summary>
+		/// Creates a new calendar or updates the name and color of an existing one.
+		/// </summary>
+		/// <param name="calendar">The calendar to create/update</param>
+		/// <exception cref="System.ArgumentException">Calendar does not exist on device or is read-only</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public async Task AddOrUpdateCalendarAsync(Calendar calendar)
+		{
+			Calendar existing = null;
+
+			if (!string.IsNullOrEmpty(calendar.ExternalID))
+			{
+				existing = await GetCalendarByIdAsync(calendar.ExternalID).ConfigureAwait(false);
+			}			
+
+			if (!existing.CanEditCalendar)
```

Comment:
`AddOrUpdateCalendarAsync` normally gets called with an empty `ExternalID` in order to create a new calendar.

Looks like doing so here would throw a `NullReferenceException` here due to `existing` not getting initialized.

If Tizen doesn't support creating new calendars, might want to instead handle the empty `ExternalID` scenario and throw an appropriate exception for that (e.g. `PlatformNotSupportedException`?).